llvm/test/CodeGen/NVPTX/and-or-setcc.ll

```diff
@@ -0,0 +1,45 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc < %s -mtriple=nvptx64 | FileCheck %s
```

Comment:
Considering that the optimization is generic, perhaps it would make sense to use `llc -stop-after=finalize-isel` and move the test under CodeGen/Generic?

Reply:
Could you point me at a good example of a test of this form? I took a quick glance at CodeGen/Generic and don't see any tests that seem to be testing DAG Combiner cases like this, or using `-stop-after=finalize-isel`.

Reply:
We seem to have some tests using `-print-after=finalize-isel`, but that gives about the same output as -stop-after, just with more work.

https://github.com/llvm/llvm-project/blob/3192ecfa89a48b5f56ff36956abe7e84327ced5d/llvm/test/CodeGen/Generic/MachineBranchProb.ll#L1

Reply:
I think the current method of testing makes more sense. The output at that point will still contain target-specific opcodes and be subject to target-specific legalization, so writing FileCheck patterns that are generic enough to cover all targets will be quite difficult. Even if I can manage it, they won't be as clear and easy to automatically update as they are now. Relying on debug printing also seems somewhat brittle.

 It seems like fairly standard practice to test generic DAGCombiner changes with checks on a specific target. Of the last functional 10 changes in DAGCombiner I checked, all added or modified target specific tests while none modified CodeGen/Generic.

Reply:
Fair enough. In that case I would suggest adding another test for a common platform like x86, so it does get wider test exposure.

Reply:
Sounds good, I've added an X86 testcase